constantine/math/polynomials/polynomials.nim

```diff
@@ -143,6 +143,32 @@ type InvDiffArrayKind* = enum
   kArrayMinus
   kMinusArray
 
+# How do we do modulo? Field.fromBigInt()?
+func sum*(r: var PolynomialCoef, f,g: PolynomialCoef)=
+  for i in  0..<r.coefs.len:
+    r.coefs[i].sum(f.coefs[i],g.coefs[i])
+
+func `+=`*(f: var PolynomialCoef, g: PolynomialCoef) =
+  for i in 0 ..< f.coefs.len:
+    f.coefs[i] += g.coefs[i]
+
+func diff*(r: var PolynomialCoef, f,g: PolynomialCoef)=
+  for i in  0..<r.coefs.len:
+    r.coefs[i].diff(f.coefs[i],g.coefs[i])
+
+func `-=`*(f: var PolynomialCoef, g: PolynomialCoef) =
+  for i in 0 ..< f.coefs.len:
+    f.coefs[i] -= g.coefs[i]
+
+func prod*[N,F](r: var PolynomialCoef[N, F], s: F, f: PolynomialCoef[N, F])=
```

Comment:
This is incorrect for polynomials in coefficient form.

`(a + bx + cx²) * (u + vx + wx²)` is not `au + bvx + cwx²`

Reply:
Agnish mentioned once that these are redundant as already there are methods for `PolynomialEvals`. Can you confirm? Because, from what I see multiplying here is of 2 polynomial coefficients and in the `PolynomialEvals` one , there is scalar getting multiplied to whole evaluated form.

Reply:
EIP-4844 focused heavily on polynomial in evaluation form because they are asymptotically more efficient than polynomial in coefficient form.

- Multiplication in evaluation form is O(n) with n the number of coefficient/evaluation.
- Naive multiplication in coefficient form is O(n²).
- Efficient multiplication in coefficient form would be O(n log n) with `n log n FFT` + `O(n) eval multiplication` + `n log n inverse FFT`

so I'm surprised if this is actually necessary in the spec?